File: app/src/main/java/com/annapol04/munchkin/engine/FakeMatch.java
package com.annapol04.munchkin.engine;

import com.annapol04.munchkin.R;
import com.annapol04.munchkin.data.EventRepository;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.Random;

import javax.inject.Inject;
import javax.inject.Named;

public class FakeMatch extends Match {
    @Inject
    public FakeMatch(Game game, @Named("myself") Player myself, EventRepository eventRepository) {
        super(game, myself, eventRepository);
    }

    @Override
    public void start(int amountOfPlayers) {
        super.start(amountOfPlayers);

        for (int i = 0; i < amountOfPlayers - 1; i++)
            eventRepository.push(new Event(Scope.GAME, Action.JOIN_PLAYER, 0, i));
    }

    @Override
    protected void namingRound() {
        super.namingRound();

        List<String> names = new ArrayList<>(Arrays.asList("Helga", "Cannabiene"));

        List<Event> events = new ArrayList<>((amountOfPlayers - 1) * 2);

        for (int i = 1; i < amountOfPlayers; i++) {
            events.add(new Event(current.getScope(), Action.NAME_PLAYER, R.string.ev_join_player, names.get(i - 1)));
            events.add(new Event(current.getScope(), Action.HAND_OVER_TOKEN, 0, players.get((i + 1) % players.size()).getScope().ordinal()));
        }

        eventRepository.push(events);
    }

    @Override
    public void handOverToken(Scope scope, int playerNr) throws IllegalEngineStateException {
        super.handOverToken(scope, playerNr);

        if (current != myself) {
            if (state == State.HAND_CARDS)
                drawInitialHandcards();
        }
    }
}

File: app/src/main/java/com/annapol04/munchkin/engine/FakeMatch.kt
package com.annapol04.munchkin.engine

import com.annapol04.munchkin.R
import com.annapol04.munchkin.data.EventRepository

import java.util.ArrayList
import java.util.Arrays

import javax.inject.Inject
import javax.inject.Named

class FakeMatch @Inject
constructor(desk: Desk,
            @Named("myself") myself: Player,
            eventRepository: EventRepository)
    : Match(desk, myself, eventRepository) {

    override fun start(amountOfPlayers: Int) {
        super.start(amountOfPlayers)

        for (i in 0 until amountOfPlayers - 1)
            eventRepository.push(Event(Scope.GAME, Action.JOIN_PLAYER, 0, i))
    }

    override fun namingRound() {
        super.namingRound()

        val names = ArrayList(Arrays.asList("Helga", "Cannabiene"))

        val events = ArrayList<Event>((amountOfPlayers - 1) * 2)

        for (i in 1..amountOfPlayers-1) {
            events.add(Event(players.value[i].scope, Action.NAME_PLAYER, R.string.ev_join_player, names[i - 1]))
            events.add(Event(players.value[i].scope, Action.HAND_OVER_TOKEN, 0, players_.value[(i + 1) % players_.value.size].scope.ordinal))
        }

        eventRepository.push(events)
    }

    @Throws(IllegalEngineStateException::class)
    override fun handOverToken(scope: Scope, playerNr: Int) {
        super.handOverToken(scope, playerNr)

        if (current != myself && areAllPlayersNamed()) {
            if (state == Match.State.STARTED)
                drawInitialHandcards()
        }
    }
}
